cluster-autoscaler/core/filter_out_schedulable.go

```diff
@@ -111,14 +111,37 @@ func (filterOutSchedulablePodListProcessor) Process(
 		unschedulablePodsToHelp = unschedulablePods
 	}
 
-	metrics.UpdateDurationFromStart(metrics.FilterOutSchedulable, filterOutSchedulableStart)
-
 	if len(unschedulablePodsToHelp) != len(unschedulablePods) {
 		klog.V(2).Info("Schedulable pods present")
-		context.ProcessorCallbacks.DisableScaleDownForLoop()
+
+		// index pods and check the difference between these 2 lists,
+		// DisableScaleDownForLoop when the pod is newer than 1h. This to avoid an unlimited scale down forbidden
+		toIndexPods, toCheckPods := unschedulablePods, unschedulablePodsToHelp
+		if len(unschedulablePodsToHelp) < len(unschedulablePods) {
+			toCheckPods, toIndexPods = unschedulablePods, unschedulablePodsToHelp
+		}
+
+		unschedulablePodsMap := make(map[string]struct{}, len(toIndexPods))
+		for _, po := range toIndexPods {
+			unschedulablePodsMap[po.Namespace+"/"+po.Name] = struct{}{}
+		}
+		for _, po := range toCheckPods {
+			podKey := po.Namespace + "/" + po.Name
+			if _, ok := unschedulablePodsMap[podKey]; ok {
+				continue
+			}
+			if time.Since(po.CreationTimestamp.Time) > time.Hour {
+				klog.V(2).Infof("Old schedulable pod present: %s", podKey)
+				continue
+			}
+			klog.V(2).Infof("Recent schedulable pod present, DisableScaleDownForLoop: %s", podKey)
+			context.ProcessorCallbacks.DisableScaleDownForLoop()
```

Comment:
I would create a boolean and set it here, then out of the loop I would call the `context.ProcessorCallbacks.DisableScaleDownForLoop()` only once if the boolean is true.

It could also be an opportunity to log less with only one log line out of the loop like:
```
klog.V(2).Infof("DisableScaleDownForLoop, recent schedulable pod(s) present, %s and +%d", podKey, countOfPods)
```